ste/jobStatusManager.go

```diff
@@ -53,59 +61,130 @@ func (jm *jobMgr) SendXferDoneMsg(msg xferDoneMsg) {
 }
 
 func (jm *jobMgr) ListJobSummary() common.ListJobSummaryResponse {
-	jm.jstm.listReq <- true
-	return <-jm.jstm.respChan
+	/*
+	 * Query job summary from handleStatusUpdateManager, but only if it's running.
+	 * Ideally we should never find it exited, but we play safe, else we will be blocked for ever.
+	 */
+	if !jm.jstm.doneHandleStatusUpdateManager {
+		jm.jstm.listReq <- true
+		return <-jm.jstm.respChan
+	} else {
+		return jm.jstm.js
+	}
 }
 
 func (jm *jobMgr) ResurrectSummary(js common.ListJobSummaryResponse) {
 	jm.jstm.js = js
 }
 
+// DrainXferDoneMessages() function drain all message on XferDone channel. There is risk of HandleStatusUpdateManager() already exited and we are blocked.
+// We safeguard against it by checking boolean set on exit of HandleStatusUpdateManager(). We try to make risk factor as low as we can, by calling this function only once.
+//
+// Note: This is not thread safe function. Onus on caller to handle that.
+func (jm *jobMgr) DrainXferDoneMessages() bool {
+	// Poke handleStatusUpdateManager to drain the xferDone channel, but only if it's running.
+	if !jm.jstm.doneHandleStatusUpdateManager && !jm.jstm.xferDoneDrainCalled {
+		jm.jstm.xferDoneDrainCalled = true
+		jm.jstm.drainXferDoneSignal <- struct{}{}
+		close(jm.jstm.xferDone)
+
+		// Wait for handleStatusUpdateManager while it drains messages from the xferDone channel.
+		<-jm.jstm.xferDoneDrainedSignal
+
+		return true
+	} else {
+		jm.Log(pipeline.LogError, fmt.Sprintf("DrainXferDoneMessages already called and its status: %s", common.IffString(jm.jstm.xferDoneDrained, "Success", "Running")))
+		return false
+	}
+}
+
+func (jm *jobMgr) CleanupJobStatusMgr() {
```

Comment:
@nitin-deamon What is the purpose of this function? I don't see it being used anywhere

Reply:
The change has been reverted.